client-side/src/stories/genericInput.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import TextField from '@mui/material/TextField';
+import InputAdornment from '@mui/material/InputAdornment';
+import '../style/genericInput.scss';
+
+const GenericInput = ({ label, type='text', value, onChange, size='medium', width='20%',
+   icon: Icon, validation, ...rest }) => {
+  const handleChange = (e) => {
+    if (onChange) {
+      onChange(e.target.value);
+    }
+  };
+
+  let error = false;
+  let helperText = '';
+
+  if (validation && typeof validation === 'function') {
```

Comment:
we can't write code in the root of the component, you should put it inside some function